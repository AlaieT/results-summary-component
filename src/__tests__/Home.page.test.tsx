import React from "react";
import { render, cleanup } from "@testing-library/react";

import Home from "../pages/home.page";

afterEach(cleanup);

describe("Home", () => {
  const props = {
    bestOf: 100,
    categoryScore: [
      { category: "Memory", score: 55 },
      { category: "Reaction", score: 59 },
      { category: "Verbal", score: 27 },
      { category: "Visual", score: 43 },
    ],
  };

  describe("correct render", () => {
    it("should match snapshot", () => {
      expect(
        render(
          <Home bestOf={props.bestOf} categoryScore={props.categoryScore} />
        ).asFragment()
      ).toMatchSnapshot();
    });
  });
});
