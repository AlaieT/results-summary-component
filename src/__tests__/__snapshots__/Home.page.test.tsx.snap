// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home correct render should match snapshot 1`] = `
<DocumentFragment>
  <main>
    <div
      id="container"
    >
      <div
        id="result"
      >
        <h1>
          Your result
        </h1>
        <div
          id="score"
        >
          46
          <span>
            of 100
          </span>
        </div>
        <h2>
          Great
        </h2>
        <p>
          Your score higher then 100% of the people who have taken these tests.
        </p>
      </div>
      <div
        id="summary"
      >
        <h1>
          Summary
        </h1>
        <ul>
          <li>
            <div
              class="container Memory"
            >
              <div
                class="category"
              >
                <div
                  height="24"
                  width="24"
                />
                Memory
              </div>
              <div
                class="score"
              >
                55
                <span>
                   / 100
                </span>
              </div>
            </div>
          </li>
          <li>
            <div
              class="container Reaction"
            >
              <div
                class="category"
              >
                <div
                  height="24"
                  width="24"
                />
                Reaction
              </div>
              <div
                class="score"
              >
                59
                <span>
                   / 100
                </span>
              </div>
            </div>
          </li>
          <li>
            <div
              class="container Verbal"
            >
              <div
                class="category"
              >
                <div
                  height="24"
                  width="24"
                />
                Verbal
              </div>
              <div
                class="score"
              >
                27
                <span>
                   / 100
                </span>
              </div>
            </div>
          </li>
          <li>
            <div
              class="container Visual"
            >
              <div
                class="category"
              >
                <div
                  height="24"
                  width="24"
                />
                Visual
              </div>
              <div
                class="score"
              >
                43
                <span>
                   / 100
                </span>
              </div>
            </div>
          </li>
        </ul>
        <button
          class="button"
        >
          Continue
        </button>
      </div>
    </div>
  </main>
  <footer>
    <p>
      Challenge by 
      <a
        href="https://www.frontendmentor.io?ref=challenge"
        rel="noopener noreferrer"
        target="_blank"
      >
        Frontend Mentor
      </a>
      .  Coded by 
      <a
        href="https://www.frontendmentor.io/profile/AlaieT"
      >
        Ilya Shafeev
      </a>
      .
    </p>
  </footer>
</DocumentFragment>
`;
